python/lsst/resources/http.py

```diff
@@ -920,7 +920,11 @@ def write(self, data: bytes, overwrite: bool = True) -> None:
             raise FileExistsError(f"Remote resource {self} exists and overwrite has been disabled")
 
         # Ensure the parent directory exists.
-        self.parent().mkdir()
+        # This is only meaningful and appropriate for WebDAV, not the general
+        # HTTP case.  e.g. for S3 HTTP URLs, the underlying service has no
+        # concept of 'directories' at all
```

Comment:
```suggestion
        # concept of 'directories' at all.
```

PEP-8 requires full sentences and so periods are needed at the end. This is a general comment for your PRs so I won't annotate every comment with a proposed fix.